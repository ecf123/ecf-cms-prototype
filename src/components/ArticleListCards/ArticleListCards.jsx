import React from 'react'
import "./ArticleListCards.scss";
import iconArrow from '../../assets/arrow-upper-right-black.svg';
import "../../styles/_reset.scss";

const ArticleListCards = ({img, date, title, articleInfo, articleLinkOne, articleLinkTwo}) => {
  return (
    <div className='card'>
        <img className='card__image' src={img} alt="Article Image" />
        <p className='card__date'>{date}</p>
        <div className='card__header'>
          <h2 className='card__title'>{title}</h2>
          <img className='card__icon' src={iconArrow} alt="Arrow Icon" />
        </div>
        <p className='card__info'>{articleInfo}</p>
        <ul className='card__links'>
            <li className='card__link card__link--purple'>{articleLinkOne}</li>
            <li className='card__link card__link--darkpurple'>{articleLinkTwo}</li>
        </ul>
    </div>
  )
}

export default ArticleListCards