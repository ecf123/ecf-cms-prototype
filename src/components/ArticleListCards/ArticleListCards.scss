@use "../../styles/variables" as *;

.card {
    font-family:$font-secondary;
    width: 26.7rem;
    height: 16.7rem;
    font-style: normal;

    &__image {
        width: 100%;
        height: auto;
    }

    &__date {
        font-weight: $font-weight-regular;
        font-size: $font-size-extra-small;
        line-height: 20px;
        color: $color-teal;
    }

    &__header {
        display: flex;
        justify-content: space-between;
        align-items: flex-start;

        .card__icon {
            margin-top: 10px;
        }
    
        .card__title {
            font-weight: $font-weight-regular;
            font-size: $font-size-medium;
            line-height: 32px;
            color: $color-black;
        }

    }


    &__info {
        font-weight: $font-weight-regular;
        font-size: $font-size-small;
        line-height: 24px;
        color: $color-teal;
    }

    &__links {
        font-weight: $font-weight-regular;
        font-size: $font-size-extra-small;
        line-height: 20px;
    }


    &__link {
        display: inline;
        margin: 0 10px;
        background-color: #F8F9FC;
        border-radius: 10px;
        padding: 0 5px;

        &--purple{
            color: $color-purple;
        }
        &--darkpurple{
            color: $color-purple-dark;
        }
    }
}