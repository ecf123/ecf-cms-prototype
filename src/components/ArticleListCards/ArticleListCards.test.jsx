import { render, screen } from "@testing-library/react";
import { customRender } from "../../utils/testUtils";

import ArticleListCards from "./ArticleListCards";

it("should render the card image", () => {
  //Arrange
  render(<ArticleListCards />);
  //Act
  const image = screen.getByAltText("Article Image");
  //Assert
  expect(image).toBeInTheDocument();
});

it("should display the date of the card on the screen", () => {
  //Arrange
  render(<ArticleListCards date="22/01/2022" />);
  //Act
  const date = screen.queryByText("22/01/2022");
  //Assert
  expect(date).toBeInTheDocument();
});

it("should display the header of the card on the screen", () => {
  //Arrange
  render(<ArticleListCards title="Test Title" />);
  //Act
  const header = screen.queryByText("Test Title");
  //Assert
  expect(header).toBeInTheDocument();
});

it("should display the links on the screen", () => {
  //Arrange
  render(
    <ArticleListCards
      articleLinkOne="Test Link One"
      articleLinkTwo="Test Link Two"
    />
  );
  //Act
  const link1 = screen.getByText("Test Link One");
  const link2 = screen.getByText("Test Link Two");
  //Assert
  expect(link1).toBeInTheDocument();
  expect(link2).toBeInTheDocument();
});

it("should display the article paragraph on the screen", () => {
  //Arrange
  render(<ArticleListCards articleInfo="Test article" />);
  //Act
  const paragraph = screen.getByText("Test article");
  //Assert
  expect(paragraph).toBeInTheDocument();
});

it("should render the arrow icon", () => {
  //Arrange
  render(<ArticleListCards />);
  //Act
  const image = screen.getByAltText("Arrow Icon");
  //Assert
  expect(image).toBeInTheDocument();
});

it("Should render layout with children", () => {
  const { container } = customRender(
    <ArticleListCards
      date="22/09/2022"
      title="Test Title"
      articleInfo="Test article"
      articleLinkOne="test link"
      articleLinkTwo="test link"
    />
  );
  expect(container).toMatchSnapshot();
});
