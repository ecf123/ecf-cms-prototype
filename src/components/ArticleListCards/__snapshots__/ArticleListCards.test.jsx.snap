// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render layout with children 1`] = `undefined`;
